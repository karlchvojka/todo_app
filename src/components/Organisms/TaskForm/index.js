import React, { useState, useEffect } from 'react';
import './index.scss';

import Datepicker from '../DatePicker'

function onDatePick(e) {
  e.preventDefault();
  // DO STUFF
  // 1. Show date DatePicker

  // 2. return back the date selected
}

function TaskForm() {
  const [picker, setPicker] = useState({});


  return (
    <section className="taskForm">
      <form action="/newtweet" method="POST">
        <div className="titleWrap">
          <label>Task Title:<br/>
            <input type="text" name="taskTitle" />
          </label>
        </div>

        <div className="dateWrap">
          <div className="startDate">
            <label>Start Date:<br/>
              <input type="text" name="startDate" onClick={onDatePick}/>
            </label>
          </div>

          <div className="endDate">
            <label>End Date:<br/>
              <input type="text" name="endDate" />
            </label>
          </div>
        </div>

        <div className="descWrap">
          <label>Description:<br/>
            <textarea id="textAreaMain" name="taskText"></textarea>
          </label>
        </div>

        <input type="submit" value="tweet" />
        </form>
      <Datepicker chooseDate={setPicker} />
      </section>
  )
}

export default TaskForm
