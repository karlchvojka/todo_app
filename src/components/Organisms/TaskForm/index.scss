@import '../../../App.scss';

.taskForm {
  box-shadow: 2px 1px 8px 5px rgba(48,191,242,0.75);
  height:100%;
  margin-bottom:25px;
  padding:20px;
  position:relative;
  width:calc(100% - 40px);

  .dateWrap {
    display:grid;
    grid-template-columns: 50% 50%;
    margin-top:20px;
  }

  .descWrap {
    margin-top:20px;
  }
}

.hidden{
  display: none;
}


.show{
  display: block;
}
.datePicker {
  background-color: #000;
  box-shadow: 2px 1px 8px 5px rgba(48,191,242,0.25);
  padding:5px;
  height: auto;
  width: 200px;

  .monthWrap {
    p {
      font-size: 12px;
      text-align:center;
      margin-bottom:5px;
      margin-top: 10px;
    }
  }

  .daysWrap {
    display: grid;
    grid-template-columns: repeat(7, 1fr);
    grid-template-rows: auto;

    .day {
      border: 1px solid $midBlue;
      margin: 2px;
      padding: 2px;

      &:hover {
        cursor: pointer;
      }

      p {
        font-size:12px;
        margin: 0px;
        text-align:center;
      }

      &.selected {
        background: $midBlue;
      }
    }
  }
}
