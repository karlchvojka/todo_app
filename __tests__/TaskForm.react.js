import React from 'react';
import {render} from '@testing-library/react';
import { create, renderer } from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

import TaskForm from '../src/components/Organisms/TaskForm';
import Datepicker from '../src/components/Organisms/DatePicker';

describe("The Task Form component", () => {
  test('should render properly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<TaskForm />);
    const result = renderer.getRenderOutput();
    expect(result.type).toBe('section');
    expect(result.props.children).toEqual([
      <section className="taskForm">
        <form action="/newtweet" method="POST">
          <input type="text" name="title" />
          <textarea id="textAreaMain" name="text"></textarea>
          <input type="submit" value="tweet" />
        </form>
        <Datepicker />
      </section>
    ]);
  })
});
