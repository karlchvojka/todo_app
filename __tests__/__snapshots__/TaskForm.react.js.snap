// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Task Form component should render properly 1`] = `
<section
  className="taskForm"
>
  <form
    action="/newtweet"
    method="POST"
  >
    <input
      name="title"
      type="text"
    />
    <textarea
      id="textAreaMain"
      name="text"
    />
    <input
      type="submit"
      value="tweet"
    />
  </form>
  <section
    className="datepicker"
  >
    <div
      className="day"
    >
      <p>
        1
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        2
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        3
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        4
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        5
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        6
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        7
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        8
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        9
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        10
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        11
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        12
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        13
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        14
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        15
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        16
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        17
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        18
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        19
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        20
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        21
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        22
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        23
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        24
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        25
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        26
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        27
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        28
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        29
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        30
      </p>
    </div>
    <div
      className="day"
    >
      <p>
        31
      </p>
    </div>
  </section>
</section>
`;
